apps/gateway/src/middlewares/models/gemini.ts

```diff
@@ -86,7 +87,40 @@ export abstract class GeminiModelAdapter extends ModelAdapterBase {
   }
 }
 
-export abstract class Gemini3ModelAdapter extends GeminiModelAdapter {}
+export abstract class Gemini3ModelAdapter extends GeminiModelAdapter {
+  transformPrompt(prompt: LanguageModelV3Prompt): LanguageModelV3Prompt {
+    return prompt.map((message) => {
+      if (message.role === "assistant" && Array.isArray(message.content)) {
+        let hasInjected = false;
+        return {
+          ...message,
+          content: message.content.map((part) => {
+            if (part.type === "tool-call") {
+              if (hasInjected) {
+                return part;
+              }
+              const thoughtSignature =
```

Comment:
1/ If the thoughtSignature exists, we should not modify anything, just pass the original object through.
2/ If we inject a thoughtSignature, we should log this so we can later show this in the console, since it can lead to suboptimal model performance.
3/ To confirm that if no provider specific behavior is implemented, passthrough is the general behavior for extra_content (meaning it is not stripped along the way)